interactors/addressNonceHandler.go

```diff
@@ -0,0 +1,109 @@
+package interactors
+
+import (
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	erdgoCore "github.com/ElrondNetwork/elrond-sdk-erdgo/core"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+)
+
+// addressNonceHandler is the handler used for one address. It is able to handle the current
+// nonce as max(current_stored_nonce, account_nonce). After each call of the getNonce function
+// the current_stored_nonce is incremented. This will prevent "nonce too low in transaction"
+// errors on the node interceptor. To prevent the "nonce too high in transaction" error,
+// a retrial mechanism is implemented. This struct is able to store all sent transactions,
+// having a function that sweeps the map in order to resend a transaction or remove them
+// because they were executed. This struct is concurrent safe.
+type addressNonceHandler struct {
+	mut                 sync.Mutex
+	address             erdgoCore.AddressHandler
+	proxy               Proxy
+	computedNonceWasSet bool
+	computedNonce       uint64
+	transactions        map[uint64]*data.Transaction
+}
+
+func newAddressNonceHandler(proxy Proxy, address erdgoCore.AddressHandler) *addressNonceHandler {
+	return &addressNonceHandler{
+		address:      address,
+		proxy:        proxy,
+		transactions: make(map[uint64]*data.Transaction),
+	}
+}
+
+func (anh *addressNonceHandler) getNonce() (uint64, error) {
+	account, err := anh.proxy.GetAccount(anh.address)
+	if err != nil {
+		return 0, err
+	}
+
+	anh.mut.Lock()
+	defer anh.mut.Unlock()
+
+	if !anh.computedNonceWasSet {
+		anh.computedNonce = account.Nonce
+		anh.computedNonceWasSet = true
+
+		return anh.computedNonce, nil
+	}
+
+	anh.computedNonce++
+
+	return core.MaxUint64(anh.computedNonce, account.Nonce), nil
+}
+
+func (anh *addressNonceHandler) reSendTransactionsIfRequired() error {
+	account, err := anh.proxy.GetAccount(anh.address)
+	if err != nil {
+		return err
+	}
+
+	anh.mut.Lock()
+	if account.Nonce == anh.computedNonce {
```

Comment:
can we get in the situation where transactions end up in the pool but don't get executed in time and this condition is met and resets the transactions list? I guess if we use getNonce correctly it won't happen since the `computedNonce` is incremented